File: MovieProject/MovieProject.Web/Modules/MovieDB/Movie/MovieForm.cs
using MovieProject.Modules.MovieDB.Movie;
using Serenity.ComponentModel;

namespace MovieProject.MovieDB.Forms;

[FormScript("MovieDB.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieForm
{
    public string Title { get; set; }
    public string Image { get; set; }
    public string Descrition { get; set; }
    public MovieKind Kind { get; set; }
    //public int GenreId { get; set; }
    public List<int> GenreList { get; set; }
    [DisplayName("Cast"), MovieCastEditor, IgnoreName]
    public List<MovieCastRow> CastList { get; set; }


}

File: MovieProject/MovieProject.Web/Modules/ServerTypes/MovieDB/MovieForm.ts
import { MovieCastEditor } from "@/MovieDB/MovieCast/MovieCastEditor";
import { StringEditor, ImageUploadEditor, EnumEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { MovieKind } from "../Modules/MovieDB.Movie.MovieKind";

export interface MovieForm {
    Title: StringEditor;
    Image: ImageUploadEditor;
    Descrition: StringEditor;
    Kind: EnumEditor;
    GenreList: LookupEditor;
    CastList: MovieCastEditor;
}

export class MovieForm extends PrefixedContext {
    static readonly formKey = 'MovieDB.Movie';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieForm.init)  {
            MovieForm.init = true;

            var w0 = StringEditor;
            var w1 = ImageUploadEditor;
            var w2 = EnumEditor;
            var w3 = LookupEditor;
            var w4 = MovieCastEditor;

            initFormType(MovieForm, [
                'Title', w0,
                'Image', w1,
                'Descrition', w0,
                'Kind', w2,
                'GenreList', w3,
                'CastList', w4
            ]);
        }
    }
}

[MovieKind]; // referenced types